### PR TITLE
Tendermint refactor

### PR DESCRIPTION
**Pull request by patiee**

### CODE REVIEW

File: codec.proto [hunk 1]
```diff
@@ -9,44 +9,50 @@ import "google/protobuf/descriptor.proto";
 
 // used in Vote
 enum SignedMsgType {
-  option (gogoproto.goproto_enum_stringer)  = true;
-  option (gogoproto.goproto_enum_prefix)    = false;
+  option(gogoproto.goproto_enum_stringer) = true; option(gogoproto.goproto_enum_prefix) = false;
```
Comment (lavabyrd) [inline codec.proto:12]: weird that this is being pushed onto one line (same on line 24), is this just formatting?

Comment (lavabyrd) [reply codec.proto:12]: oh is it because there's no space between `options` and the `(gogoproto` anymore? Is this intended?

**Comment by lavabyrd** [comment]
Is this just for formatting btw? Or is there something fundamental in this that helps with the subgraph etc? Any Jira link also? :)

File: codec.proto (hunk 1 above)
Comment (patiee) [reply codec.proto:12]: I was testing `clang-format`, you can find command in readme :)

**Comment by patiee** [comment]
> Is this just for formatting btw? Or is there something fundamental in this that helps with the subgraph etc? Any Jira link also? :)

Yes just refactoring, I separeted cosmos changes from tendermint, but now we have consistent definitions everywhere.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 .DS_Store
+node_modules
```
Comment (lavabyrd) [inline .gitignore:2]: whats this for? I don't think there's any node modules in this repo?

File: codec.proto [hunk 3]
```diff
@@ -9,44 +9,50 @@ import "google/protobuf/descriptor.proto";
...
 }
 
-message EventData {
-  Event       event = 1;
-  EventBlock  block = 2;
+// Reward
+message Reward {
+  string amount    = 1;
+  string validator = 2;
 }
```
Comment (pienkowb) [inline codec.proto:49]: The `Reward` type is not needed in the protobuf, we should keep it only in the example subgraph schema.

File: codec.proto [hunk 4]
```diff
@@ -9,44 +9,50 @@ import "google/protobuf/descriptor.proto";
...
-  option (gogoproto.goproto_enum_prefix)    = false;
+  option(gogoproto.goproto_enum_stringer) = true; option(gogoproto.goproto_enum_prefix) = false;
 
   BLOCK_ID_FLAG_UNKNOWN = 0 [(gogoproto.enumvalue_customname) = "BlockIDFlagUnknown"];
   BLOCK_ID_FLAG_ABSENT  = 1 [(gogoproto.enumvalue_customname) = "BlockIDFlagAbsent"];
   BLOCK_ID_FLAG_COMMIT  = 2 [(gogoproto.enumvalue_customname) = "BlockIDFlagCommit"];
   BLOCK_ID_FLAG_NIL     = 3 [(gogoproto.enumvalue_customname) = "BlockIDFlagNil"];
 }
 
+// EventData
```
Comment (pienkowb) [inline codec.proto:32]: This comment does not convey any more information than the message name, we can remove it.

File: codec.proto [hunk 5]
```diff
@@ -9,44 +9,50 @@ import "google/protobuf/descriptor.proto";
...
+message EventData {
+  Event      event = 1;
+  EventBlock block = 2;
+}
+
+// EventList
```
Comment (pienkowb) [inline codec.proto:38]: Same here.

File: codec.proto [hunk 6]
```diff
@@ -9,44 +9,50 @@ import "google/protobuf/descriptor.proto";
...
 message EventList {
```
Comment (pienkowb) [inline codec.proto:39]: `EventList` is the entry point to the whole structure, it should be the first definition (right after enums).

File: codec.proto [hunk 7]
```diff
@@ -60,21 +66,22 @@ message BlockParams {
...
+  repeated CommitSig signatures = 4 [(gogoproto.nullable) = false];
 }
 
 message CommitSig {
   BlockIDFlag block_id_flag     = 1; // must match those in the enum above
-  bytes       validator_address = 2; // hash 256 20 bytes - this is the first 20 characters of a 32-byte key - SHA256(pubkey)[:20]
-  Timestamp   timestamp         = 3 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
-  bytes       signature         = 4; // length should be > 0 and < 64
+  bytes       validator_address = 2; // hash 256 20 bytes - this is the first 20 characters of a 32-byte key -
+                                     // SHA256(pubkey)[:20]
```
Comment (pienkowb) [inline codec.proto:78]: Comments broken into multiple lines do not translate well to the generated files ([example](https://github.com/figment-networks/graph-node/commit/e26516e5eb5cbbe8a9f636788bee77b5e7442947#diff-25193c25bf9f48bedbed65facbcd45bbd38a39b01373bee892b4c4be1176bda0R64-R67)). Maybe we can tweak the formatter (or make the comment shorter)?

**Review by pienkowb — COMMENTED** [review]
I've added some comments. There's no rush with this - the changes are backward-compatible so we can tweak the formatting after the release.

**Review by pienkowb — APPROVED** [review]
Looks good 👍 Let's make sure that we roll this out along with the changes to graph-ts and graph-node.

**Pull request merged**